nucliadb/nucliadb/ingest/orm/entities.py

```diff
@@ -0,0 +1,261 @@
+from nucliadb.ingest.maindb.keys import (
+    KB_DELETED_ENTITIES_GROUPS,
+    KB_ENTITIES,
+    KB_ENTITIES_GROUP,
+)
+from nucliadb.ingest.orm import knowledgebox
+
+
+class EntitiesManager:
+    def __init__(self, knowledgebox: "knowledgebox.Knowledgebox", txn):
+        self.kb = knowledgebox
+        self.txn = txn
+        self.kbid = self.kb.kbid
+
+    async def get_entities(self, entities: GetEntitiesResponse):
+        async for group, eg in self.entitiesgroups_iterator(exclude_deleted=True):
+            entities.groups[group].CopyFrom(eg)
+
+    async def get_entitiesgroup(
+        self, group: str, entitiesgroup: GetEntitiesGroupResponse
+    ):
+        deleted_groups = await self.get_deleted_entitiesgroups()
+        if group in deleted_groups:
+            return
+        eg = await self.get_entitiesgroup_inner(group)
+        if eg is not None:
+            entitiesgroup.group.CopyFrom(eg)
+
+    async def set_entities(self, group: str, entities: EntitiesGroup):
+        new_entities = EntitiesGroup()
+        stored_entities = await self.get_entitiesgroup_inner(group)
+
+        if stored_entities is None:
+            new_entities = entities
+        else:
+            merged = {}
+
+            for key, entity in stored_entities.entities.items():
+                if key not in entities.entities:
+                    entity.status = Entity.DiffStatus.DELETED
+                else:
+                    entity.status = entities.entities[key].status
+                merged[key] = entity
+
+            # overwrite stored entities with new ones
+            merged.update(entities.entities)
+
+            new_entities.MergeFrom(
+                EntitiesGroup(
+                    entities=merged,
+                    title=entities.title,
+                    color=entities.color,
+                    custom=entities.custom,
+                )
+            )
+
+        entities_key = KB_ENTITIES_GROUP.format(kbid=self.kbid, id=group)
+        await self.txn.set(entities_key, new_entities.SerializeToString())
+        # TODO: properly index new entities
+        await self.index_entities(group, new_entities)
+
+    async def set_entities_force(self, group: str, entities: EntitiesGroup):
+        entities_key = KB_ENTITIES_GROUP.format(kbid=self.kbid, id=group)
+        await self.txn.set(entities_key, entities.SerializeToString())
+
+    async def del_entities(self, group: str):
+        entities_key = KB_ENTITIES_GROUP.format(kbid=self.kbid, id=group)
+        await self.txn.delete(entities_key)
+
+        deleted_groups_key = KB_DELETED_ENTITIES_GROUPS.format(kbid=self.kbid)
+        payload = await self.txn.get(deleted_groups_key)
+        deg = DeletedEntitiesGroups()
+        if payload is not None:
+            deg.ParseFromString(payload)
+        if group not in deg.entities_groups:
+            deg.entities_groups.append(group)
+        await self.txn.set(deleted_groups_key, deg.SerializeToString())
```

Comment:
We should update the relations index upon deletions too, right?

Reply:
Mm... Good catch! I'm not sure about it. 

If the entities group is from the system, it definitely shouldn't be deleted from the index and we'll provide a view to the user. But for entities groups created by the user, might have sense to remove them from the index. @bloodbare what do you think?

If we decide to remove them, we have to be sure to only delete user-defined entities groups.

Reply:
Was there a decision on this front?

Reply:
We can always add this later. For me, we can move on with this PR